src/builtins.rs

```diff
@@ -167,12 +179,89 @@ lazy_static! {
+                                | (DataTypeInformation::Integer { .. }, DataTypeInformation::Integer {signed : true, size, ..}) if is_exponent_positive_literal => {
+                                    //Convert both to minimum dint
+                                    let element_type = get_bigger_type(element_type, dint_type, annotator.index);
+                                    let exponant_type = if *size <= DINT_SIZE {
+                                        udint_type
+                                    } else {
+                                        exponant_type
+                                    };
+                                    (element_type.get_name(), exponant_type.get_name())
+                                },
+                                //If left is real, then if right is int call powi
+                                (_, DataTypeInformation::Integer {size, ..}) => {
+                                    //Convert the exponent to minimum DINT
+                                    let target_type = get_bigger_type(element_type, real_type, annotator.index);
+
+                                    // For integer powers, only a 32 bit integer is allowed
+                                    let exponant_type = if *size <= DINT_SIZE {
+                                        dint_type
+                                    } else {
+                                        //For bigger types, we move to the target type (Effectively always LREAL)
+                                        get_bigger_type(target_type, exponant_type, annotator.index)
+                                    };
+                                    let target_type = get_bigger_type(target_type, exponant_type, annotator.index);
+                                    (target_type.get_name(), exponant_type.get_name())
+                                },
+                                //If right is real convert to common real type and call powf
+                                _ => {
+                                    //Convert left and right to minimum REAL
+                                    let target_type = get_bigger_type(
+                                        get_bigger_type(element_type, exponant_type, annotator.index), real_type, annotator.index);
+                                    (target_type.get_name(), target_type.get_name())
+                                }
+                            };
+                            let mut generics_candidates = HashMap::new();
+                            generics_candidates.insert("U".to_string(), vec![element_type.to_string()]);
+                            generics_candidates.insert("V".to_string(), vec![exponant_type.to_string()]);
+                            annotator.update_generic_call_statement(generics_candidates, "EXPT", operator, parameters, ctx)
+                        }
```

Comment:
is an else here unreachable? can we add an !impossible with explanation?

Reply:
I added an unreachable here, but I'm not sure this is the best way.
_Theoratically_ these types exist, and if not it's the resolver's / validator's job to say so.